inject.js

```diff
@@ -408,24 +419,34 @@ function setupListener() {
         log("Speed event propagation blocked", 4);
         event.stopImmediatePropagation();
       }
-      var video = event.target;
-      var speedIndicator = video.vsc.speedIndicator;
-      var src = video.currentSrc;
-      var speed = video.playbackRate.toFixed(2);
-
-      log("Playback rate changed to " + speed, 4);
-
-      log("Updating controller with new speed", 5);
-      speedIndicator.textContent = speed;
-      tc.settings.speeds[src] = speed;
-      log("Storing lastSpeed in settings for the rememberSpeed feature", 5);
-      tc.settings.lastSpeed = speed;
-      log("Syncing chrome settings for lastSpeed", 5);
-      chrome.storage.sync.set({ lastSpeed: speed }, function () {
-        log("Speed setting saved: " + speed, 5);
-      });
-      // show the controller for 1000ms if it's hidden.
-      runAction("blink", document, null, null);
+      var controller = event.target.parentElement.querySelector(
+        ".vsc-controller"
+      );
+      var video = controller.parentElement.querySelector("video");
```

Comment:
Why the gymnastics to select parentElement and then traverse down? Ditto for controller?